source/core/TDCodecVia.cpp

```diff
@@ -1770,6 +1770,7 @@ TDViaFormatter::TDViaFormatter(StringRef string, Boolean quoteOnTopString, Int32
         _options._bEscapeStrings = false;
         _options._fmt = formatVIA;
     } else if (format->ComparePrefixCStr(formatJSON._name)) {
+        _options._precision = 17;
```

Comment:
2.7999999999999998 is the value that is currently returned in CG for 2.8 in FlattenToJSON. I'll see if 16 returns the same value.